### PR TITLE
breaking out role resources

### PR DESCRIPTION
**Pull request by brendanmcfarland**
First shot at starting to limit the resources when creating a spotfleet as part of the quest for native spotfleet [tagging](https://github.com/mapbox/ecs-cluster/pull/480)

cc: @jakepruitt @rclark @kellyoung

### CODE REVIEW

File: lib/build.js [hunk 1]
```diff
@@ -193,11 +193,21 @@ function build(params) {
...
+                  Action: [
+                    'iam:ListRoles',
+                    'iam:PassRole'
+                  ],
+                  Resource: params.Properties.SpotFleetRequestConfigData.IamFleetRole
+                },
+                {
+                  Effect: 'Allow',
+                  Action: ['iam:ListInstanceProfiles'],
+                  Resource: params.Properties.SpotFleetRequestConfigData.InstanceProfile
```
Comment (rclark) [inline lib/build.js]: I thought you were gonna have to pull this out of LaunchSpecifications?

Comment (brendanmcfarland) [reply lib/build.js]: I realized you can pass anything to magic-cfn-functions in the SpotFleetRequestConfigData, and seemed simpler to pass all the roles the same way.

Comment (rclark) [reply lib/build.js]: Before you made this change, the parameters that you had to pass to this magic function were _exactly the same_ as what you would pass to a real-life spot fleet resource on cloudformation. After this change, you have introduced one little exception to that rule. Is that a good idea?

Comment (brendanmcfarland) [reply lib/build.js]: >After this change, you have introduced one little exception to that rule. Is that a good idea?

When you put it that way, it sure doesn't sound like a good idea. I'll update to point at 
> params.Properties.SpotFleetRequestConfigData.LaunchSpecifications[0].IamInstanceProfile

Which I think should accomplish the same thing and not add a new exception to the params.

**Pull request merged**